### PR TITLE
Add 'checks' Make and CI tasks

### PR DESCRIPTION
**Pull request by silverwind**
Introduce new "sanity" targets that perform tasks that we've been piling onto the linting tasks. This will make the linter tasks faster and hopefully encourage some users to use them locally.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
Not sure I like the "sanity" name, guess we need something more fitting.

**Comment by silverwind** [comment]
Renamed to "checks".

**Comment by silverwind** [comment]
I'm not sure if `misspell-check` ever ran on the CI, but I've added it now.

**Comment by silverwind** [comment]
Mispell successfully ran. I think this is ready for merge.

File: Makefile [hunk 1]
```diff
@@ -290,17 +290,26 @@ fmt-check:
...
+lint: lint-frontend lint-backend
 
 .PHONY: lint-frontend
-lint-frontend: node_modules svg-check
+lint-frontend: node_modules
 	npx eslint web_src/js build webpack.config.js
 	npx stylelint web_src/less
 
+.PHONY: lint-backend
+lint-backend: golangci-lint revive vet
```
Comment (zeripath) [inline Makefile:314]: Would it be useful to add

```suggestion
lint-backend: checks-backend golangci-lint revive vet
```

File: Makefile [hunk 2]
```diff
@@ -290,17 +290,26 @@ fmt-check:
...
+
+.PHONY: checks-backend
+checks-backend: misspell-check test-vendor swagger-check swagger-validate
+
+.PHONY: lint
+lint: lint-frontend lint-backend
 
 .PHONY: lint-frontend
-lint-frontend: node_modules svg-check
+lint-frontend: node_modules
```
Comment (zeripath) [inline Makefile:309]: Would it be useful to add?

```suggestion
lint-frontend: checks-frontend node_modules
```

File: Makefile (hunk 1 above)
Comment (silverwind) [reply Makefile:314]: I think not, I want to separate linting from the checks to be able to run only the linters locally.

File: Makefile (hunk 2 above)
Comment (silverwind) [reply Makefile:309]: Same reason as above. I want to run linter only.

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**